sdk/Pulumi/Serialization/Attributes.cs

```diff
@@ -108,4 +108,19 @@ public ResourceTypeAttribute(string type, string? version)
             Version = version;
         }
     }
+
+    /// <summary>
+    /// Attribute used by a Pulumi Cloud Provider Package to mark
+    /// constructor parameters with a name override.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class ConstructorParameterNameOverrideAttribute : Attribute
```

Comment:
The type should be `sealed`.

Also, I wonder if a better name would be `OutputConstructorParameterAttribute`, to align with `OutputConstructorAttribute`:

```suggestion
    public sealed class OutputConstructorParameterAttribute : Attribute
```

I also wonder if this should be moved directly below `OutputConstructorAttribute` (line 76).

Reply:
I've made changes for both suggestions you've made here. Also on the topic of the attribute class being `sealed` I noticed that `ResourceTypeAttribute` is not marked as such. That's the one I was using as reference and hence didn't mark the new one as `sealed` initially. Would you like me to mark it as `sealed` as well?